### PR TITLE
Update action versions

### PR DESCRIPTION
**Pull request by CherryPerry**
@arkivanov

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,33 +16,28 @@ jobs:
...
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
-          java-version: 11
+          distribution: 'zulu'
+          java-version: '11'
       - name: Install dependencies
         if: matrix.os == 'ubuntu-latest'
         run: sudo apt-get update && sudo apt-get install libcurl4-openssl-dev libgtk-3-dev nodejs chromium-browser
-      - name: Set up cache
```
Comment (arkivanov) [inline .github/workflows/build.yml:30]: Why did you remove this?

File: .github/workflows/release-nmtc.yml [hunk 2]
```diff
@@ -15,12 +12,12 @@ jobs:
     steps:
       - id: create
         name: Create staging repository
-        uses: nexus-actions/create-nexus-staging-repo@v1
+        uses: nexus-actions/create-nexus-staging-repo@67d9c1133c2fdcb777e20c040595073e20311225
```
Comment (arkivanov) [inline .github/workflows/release-nmtc.yml:15]: Why not a stable version?

File: .github/workflows/build.yml (hunk 1 above)
Comment (CherryPerry) [reply .github/workflows/build.yml:30]: `gradle/gradle-build-action` manages caches keys by itself, better to rely on Gradle team rather than managing it ourselves.

File: .github/workflows/release-nmtc.yml (hunk 2 above)
Comment (CherryPerry) [reply .github/workflows/release-nmtc.yml:15]: I am ready to rely on well-known actions maintained by teams, but this one unfortunately is not and we are passing very critical information there.

Also it is [advice](https://github.com/nexus-actions/create-nexus-staging-repo) from the author:

```       
# replace '@main' with the latest commit sha, see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions
```

File: .github/workflows/build.yml (hunk 1 above)
Comment (arkivanov) [reply .github/workflows/build.yml:30]: I thought that GitHub Actions drops everything including cache folders by default. But if you believe it's ok then lgtm.

Comment (CherryPerry) [reply .github/workflows/build.yml:30]: [gradle/gradle-build-action](https://github.com/gradle/gradle-build-action#caching)

File: .github/workflows/release-nmtc.yml (hunk 2 above)
Comment (arkivanov) [reply .github/workflows/release-nmtc.yml:15]: TIL

File: .github/workflows/build.yml (hunk 1 above)
Comment (arkivanov) [reply .github/workflows/build.yml:30]: TIL

**Review by arkivanov — APPROVED** [review]
